src/test/java/org/cactoos/io/ZipTest.java

```diff
@@ -24,51 +24,87 @@
+import org.llorllale.cactoos.matchers.MatcherOf;
 
 /**
  * Test case for {@link Zip}.
  *
  * @since 0.29
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 public final class ZipTest {
+
     /**
      * Temporary folder.
      */
     @Rule
-    public final TemporaryFolder folder = new TemporaryFolder();
+    public final TemporaryFolder temporal = new TemporaryFolder();
 
     @Test
-    public void zip() throws Exception {
-        final Path dir = this.folder.newFolder().toPath();
-        dir.resolve("x/y").toFile().mkdirs();
-        Files.write(dir.resolve("x/y/test"), "".getBytes());
-        try (ZipInputStream input = new ZipInputStream(
-            new Zip(
-                new Directory(dir)
-            ).stream()
-        )) {
-            int cnt = 0;
-            ZipEntry entry = input.getNextEntry();
-            while (entry != null) {
-                ++cnt;
-                entry = input.getNextEntry();
-            }
-            MatcherAssert.assertThat(
-                "Can't list files in a directory represented by a path",
-                cnt,
-                // @checkstyle MagicNumber (1 line)
-                Matchers.equalTo(4)
-            );
+    public void mustZipDirectory() throws Exception {
+        final String zipname = "abc.zip";
+        final File folder = this.temporal.newFolder("abc");
+        try (OutputStream out = new OutputStreamTo(
+            new File(folder, "A.txt")
+        )
+        ) {
+            out.write("ABC".getBytes());
+        }
+        try (OutputStream out = new OutputStreamTo(
+            new File(folder, "B.txt")
+        )
+        ) {
+            out.write("EFG".getBytes());
         }
+        new Assertion<>(
+            "Must zip directory with the same directory structure",
+            new Sticky<>(
+                () -> {
+                    new LengthOf(
+                        new TeeInput(
+                            new InputOf(
+                                new Zip(
+                                    new Directory(folder)
+                                ).stream()
+                            ),
+                            new OutputTo(
+                                this.temporal.newFile(zipname)
+                            )
+                        )
+                    ).value();
+                    try (ZipFile file = new ZipFile(
+                        new File(folder.getParentFile(), zipname)
+                    )
+                    ) {
+                        return file.stream().map(ZipEntry::toString).collect(
+                            Collectors.toList()
+                        );
+                    }
+                }
+            ),
+            new MatcherOf<>(
+                sclr -> {
+                    sclr.value().containsAll(
+                        new ListOf<>(
+                            "abs\\A.txt",
```

Comment:
@fanifieiev is it really `abs`?

Reply:
@victornoel No, it is a mistake. And also I did missed `return` that's why the test did not fail. Will fix.